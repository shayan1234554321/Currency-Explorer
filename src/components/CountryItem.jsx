import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import flagImages from '../utility/flags';
import arrow from '../assets/arrow.svg';
import { SetCurrentCurrency } from '../redux/currencies/currenciesSlice';

const CountryItem = ({ country }) => {
  const {
    country_name: countryName,
    country_code: countryCode,
    currency_name: currencyName,
    currency_value: currencyValue,
    currency_code: currencyCode,
  } = country;
  const dispatch = useDispatch();
  const navigate = useNavigate();

  function handleClick() {
    const code = currencyCode.toLowerCase();
    dispatch(SetCurrentCurrency(code));
    navigate('/currency');
  }

  return (
    <button type="button" className="countryItem" onClick={handleClick}>
      <div className="visit">
        <img src={arrow} alt="" />
      </div>
      <div className="details">
        <h2>{countryName}</h2>
        <h4>{currencyName}</h4>
        <h4>
          {Math.round(currencyValue * 1000) / 1000}
          {' '}
          {currencyCode}
        </h4>
      </div>
      <div className="imageContainer">
        <img src={flagImages[countryCode]} alt="" />
      </div>
    </button>
  );
};

CountryItem.propTypes = {
  country: PropTypes.shape({
    country_name: PropTypes.string.isRequired,
    country_code: PropTypes.string.isRequired,
    currency_name: PropTypes.string.isRequired,
    currency_value: PropTypes.number.isRequired,
    currency_code: PropTypes.string.isRequired,
  }).isRequired,
};

export default CountryItem;
