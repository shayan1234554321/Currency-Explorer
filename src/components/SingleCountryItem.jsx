import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import arrow from '../assets/arrow.svg';
import { SetCurrentCurrency } from '../redux/currencies/currenciesSlice';

const SingleCountryItem = ({ country }) => {
  const {
    country_name: countryName,
    currency_value: currencyValue,
    currency_code: currencyCode,
  } = country;
  const dispatch = useDispatch();
  const handleCountryClick = () => {
    const code = currencyCode.toLowerCase();
    dispatch(SetCurrentCurrency(code));
  };

  return (
    <button type="button" className="SingleCountryItem" onClick={handleCountryClick}>
      <h4>{countryName}</h4>
      <div className="right">
        <h3>
          {Math.round(currencyValue * 1000) / 1000}
          {' '}
          {currencyCode}
        </h3>
        <img src={arrow} alt="left arrow" />
      </div>
    </button>
  );
};

SingleCountryItem.propTypes = {
  country: PropTypes.shape({
    country_name: PropTypes.string.isRequired,
    currency_value: PropTypes.number.isRequired,
    currency_code: PropTypes.string.isRequired,
  }).isRequired,
};

export default SingleCountryItem;
