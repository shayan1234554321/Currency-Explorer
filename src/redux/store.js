import { configureStore } from '@reduxjs/toolkit';
import currenciesSlice from './currencies/currenciesSlice';

const store = configureStore({
  reducer: {
    currencies: currenciesSlice,
  },
});

export default store;
