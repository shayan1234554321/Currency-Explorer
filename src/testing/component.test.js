import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import CountryItem from '../components/CountryItem';
import SingleCountryItem from '../components/SingleCountryItem';

const mockSingleCountryItem = {
  country_name: 'Pakistan',
  currency_value: 1,
  currency_code: 'pkr',
};

const mockCountryItem = {
  country_name: 'Pakistan',
  country_code: 'pk',
  currency_name: 'pakistani rupees',
  currency_value: 1,
  currency_code: 'pkr',
};

describe('Renders without crashing', (() => {
  it('CountryItem to Match Snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/currency" Component={<CountryItem country={mockCountryItem} />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SingleCountryItem to Match Snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/currency" Component={<SingleCountryItem country={mockSingleCountryItem} />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
}));
