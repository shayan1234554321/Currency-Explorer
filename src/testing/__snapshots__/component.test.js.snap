// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing CountryItem to Match Snapshot 1`] = `null`;

exports[`Renders without crashing SingleCountryItem to Match Snapshot 1`] = `null`;
